src/com/facebook/buck/ide/intellij/IjProjectCommandHelper.java

```diff
@@ -233,6 +251,75 @@ private TargetGraph getProjectGraphForIde(
         buckEventBus, cell, enableParserProfiling, executor, passedInTargets);
   }
 
+  @SuppressWarnings("unchecked")
+  private ExitCode copyGeneratedGoSource(TargetGraphAndTargets targetGraphAndTargets)
+      throws IOException, InterruptedException {
+    HashMap<String, BuildTargetSourcePath> generatedPackages = new HashMap<>();
+    ClassLoader classLoader = pluginManager.getPluginClassLoader("com.facebook.buck.features.go");
+    try {
+      Class<?> goLibraryDescriptionArgClass =
+          Class.forName("com.facebook.buck.features.go.GoLibraryDescriptionArg", true, classLoader);
+      for (TargetNode<?, ?> targetNode : targetGraphAndTargets.getTargetGraph().getNodes()) {
+        Object constructorArg = targetNode.getConstructorArg();
+        if (constructorArg.getClass() == goLibraryDescriptionArgClass) {
+          Method getPackageNameMethod = goLibraryDescriptionArgClass.getMethod("getPackageName");
+          Optional<String> packageName =
+              (Optional<String>) getPackageNameMethod.invoke(constructorArg);
+          if (packageName.isPresent()) {
+            generatedPackages.putAll(
+                ((HasSrcs) constructorArg)
+                    .getSrcs()
+                    .stream()
+                    .filter(srcPath -> srcPath instanceof BuildTargetSourcePath)
+                    .collect(
+                        Collectors.toMap(
+                            src -> packageName.get(), src -> (BuildTargetSourcePath) src)));
+          }
```

Comment:
else fail the build? Are there any valid cases when go library may have no package name?

Reply:
Yes there are, but we don't. Figuring out a proper default package name will need to load another class from go module and use the same reflection trick to call its methods.